src/core/features/course/pages/list-mod-type/list-mod-type.ts

```diff
@@ -125,19 +141,64 @@ export class CoreCourseListModTypePage implements OnInit {
             const result = await CoreCourseHelper.addHandlerDataForModules(sections, this.courseId);
 
             this.sections = result.sections;
+
+            this.canLoadMore = false;
+            this.showSectionId = 0;
+            this.showMoreActivities();
         } catch (error) {
             CoreDomUtils.showErrorModalDefault(error, 'Error getting data');
         }
     }
 
+    /**
+     * Show more activities (only used when showing all the sections at the same time).
+     *
+     * @param infiniteComplete Infinite scroll complete function. Only used from core-infinite-loading.
+     */
+    showMoreActivities(infiniteComplete?: () => void): void {
+        this.canLoadMore = false;
+
+        const sections = this.sections || [];
+        let modulesLoaded = 0;
+        let i: number;
+        for (i = this.showSectionId + 1; i < sections.length; i++) {
+            if (!sections[i].hasContent || !sections[i].modules) {
+                continue;
+            }
+
+            modulesLoaded += sections[i].modules.length;
+
+            if (modulesLoaded >= CoreCourseListModTypePage.LOAD_MORE_ACTIVITIES) {
+                break;
+            }
+        }
+
+        this.showSectionId = i;
+        this.canLoadMore = i < sections.length;
+
+        if (this.canLoadMore) {
+            // Check if any of the following sections have any content.
+            let thereAreMore = false;
+            for (i++; i < sections.length; i++) {
+                if (sections[i].hasContent && sections[i].modules && sections[i].modules?.length > 0) {
+                    thereAreMore = true;
+                    break;
+                }
+            }
+            this.canLoadMore = thereAreMore;
+        }
```

Comment:
```suggestion
        this.canLoadMore = sections.slice(i).some(section => sections[i].hasContent && sections[i].modules && sections[i].modules.length > 0);
```

If you follow my suggestion above, this will be removed. But if you don't, that's a better approach.